clang/test/Driver/sycl-device-lib-old-model.cpp

```diff
@@ -378,3 +378,40 @@
 // RUN: -Xarch_device -fsanitize=memory -### 2>&1 | FileCheck %s -check-prefix=SYCL_DEVICE_LIB_MSAN_CPU
 // SYCL_DEVICE_LIB_MSAN_CPU: llvm-link{{.*}} "-only-needed" "{{.*}}libsycl-crt.bc"
 // SYCL_DEVICE_LIB_MSAN_CPU-SAME: "{{.*}}libsycl-msan-cpu.bc"
+
+/// ###########################################################################
+/// test behavior of libsycl-tsan.bc linking when -fsanitize=thread is available
+// RUN: %clangxx -fsycl --no-offload-new-driver %s --sysroot=%S/Inputs/SYCL -fsanitize=thread -### 2>&1 \
+// RUN:   | FileCheck %s -check-prefix=SYCL_DEVICE_LIB_TSAN
+// RUN: %clangxx -fsycl --no-offload-new-driver %s --sysroot=%S/Inputs/SYCL -Xsycl-target-frontend -fsanitize=thread -### 2>&1 \
```

Comment:
Why is `-Xsycl-target-frontend` needed here?

Reply:
As we know, '-Xsycl-target-frontend' is equal to '-Xarch_device', the difference is that '-Xsycl-target-frontend' is sycl specific flag and '-Xarch_device' is supported by community. And device sanitizer needs to support both, so we add such tests here.